### ISSUE TITLE
ERROR: System.NullReferenceException: Xamarin.Forms.Core\PlatformConfiguration\iOSSpecific\Page.cs:134

### ISSUE THREAD

**Issue by wenyonk** [xamarin/Xamarin.Forms#3866]
### Description
After upgrading to the latest Xcode (10.0 10A255), VS Studio for Mac (7.6.6 build 17) and Xamarin.Forms(3.2.0.839983), I cleaned the solution, deleted the app from my devices and rebuilt and deployed the app crashed on startup with the following error:

10:02:25 - ERROR: System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.PlatformConfiguration.iOSSpecific.Page.GetSafeAreaInsets (Xamarin.Forms.BindableObject element) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\PlatformConfiguration\iOSSpecific\Page.cs:134 
  at Xamarin.Forms.PlatformConfiguration.iOSSpecific.Page+<>c.<.cctor>b__27_0 (Xamarin.Forms.BindableObject bindable, System.Object oldValue, System.Object newValue) [0x00027] in D:\a\1\s\Xamarin.Forms.Core\PlatformConfiguration\iOSSpecific\Page.cs:70 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00120] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:625 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x0015b] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:417 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0003d] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:573 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:99 
  at Core.App.InitializeComponent () [0x00023] in /Core/obj/Debug/App.xaml.g.cs:26 
  at Core.App..ctor () [0x00009] in /Core/App.xaml.cs:145 

### Steps to Reproduce
In each of the content pages xml the following code statements are present:
 **xmlns:iOSSpecific="clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core" 
    iOSSpecific:Page.UseSafeArea="true"**

1. Upgrade to the latest Xcode (10.0 10A255), VS Studio for Mac (7.6.6 build 17) and Xamarin.Forms(3.2.0.839983)
2. Clean the solution, delete the app from your devices
3. Rebuilt, deployed and run the app from the debugger
4. The app should crash upon startup 

### Expected Behavior
App should not crash

### Actual Behavior
App crashed as stated above

### Basic Information
- Version with issue: VS for Mac 7.6.6 (build 17)
- Last known good version: Previous stable version, previous stable version of Xamarin Forms
- IDE:
- Platform Target Frameworks: iOS and Android
  - iOS:  Target Deployment 9.2, Xcode 10.0
  - Android: Min API 23, Target API 26, SDK TOOLS 26.1.1, PLATFORM TOOLS 28.0.1 
  
=== Visual Studio Professional 2017 for Mac ===

Version 7.6.6 (build 17)
Installation UUID: 287748ba-1b41-4c91-8123-0c3b994d739f
Runtime:
	Mono 5.12.0.301 (2018-02/4fe3280bba1) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 512000301

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.3
	2.1.2
	2.1.1
	2.1.0
	2.0.5
	2.0.0
	1.1.1
	1.0.4
SDK: /usr/local/share/dotnet/sdk/2.1.401/Sdks
SDK Versions:
	2.1.401
	2.1.302
	2.1.301
	2.1.300
	2.1.4
	2.0.0
	1.0.3
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.3
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 10.0 (14320.25)
Build 10A255

=== Xamarin.Mac ===

Version: 4.6.0.14 (Visual Studio Professional)
Hash: 1a6e2786
Branch: 
Build date: 2018-08-28 00:12:06-0400

=== Xamarin.iOS ===

Version: 12.0.0.15 (Visual Studio Professional)
Hash: 84552a46
Branch: xcode10
Build date: 2018-09-17 21:54:33-0400

=== Xamarin.Android ===

Version: 9.0.0.20 (Visual Studio Professional)
Android SDK: /Users/kenwenyon/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		6.0 (API level 23)
		7.0 (API level 24)
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.1
SDK Build Tools Version: 27.0.3

Java SDK: /usr
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 706060017
Git revision: 822323b5f6295b271d4c9bfaca4129f90ec41f38
Build date: 2018-09-18 20:19:40+00
Build branch: release-7.6-xcode10
Xamarin extensions: a83eb4b7ad9174ac2e705f2c4257ecd15688ba97

=== Operating System ===

Mac OS X 10.13.6
Darwin 17.7.0 Darwin Kernel Version 17.7.0
    Thu Jun 21 22:53:14 PDT 2018
    root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

NuGet Package Explorer 0.2
NuGet Package Management Extensions 0.13

### Screenshots
Please see StackTrace above

**Comment by samhouts** [comment]
Would you mind attaching a small project that demonstrates this issue? Thank you!

**Comment by pit-le-rouge** [comment]
I have exactly the same problem after migrating to FORMS 3.2 and sudio for MAc up-to-date !

in debug, the program stop in the app.xaml.g.cs:

[global::Xamarin.Forms.Xaml.XamlFilePathAttribute("App.xaml")]
public partial class App : global::Xamarin.Forms.Application {
     [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Xamarin.Forms.Build.Tasks.XamlG", "0.0.0.0")]
     private void InitializeComponent() {
            global::Xamarin.Forms.Xaml.Extensions.LoadFromXaml(this, typeof(App));
     }
}

the error message "System.NullReferenceException" come just after the 
global::Xamarin.Forms.Xaml.Extensions.LoadFromXaml(this, typeof(App));

HELP !

**Comment by jalbertSyncroTech** [comment]
@samhouts just confirming that we started seeing as well this after upgrading from v3.1.0.697729 to v3.2.0.839982.

We set the UseSafeArea property via a common style we apply to our ContentPage's, regardless of the platform.  We aren't getting any issues actually running on UWP or Android, but in our unit tests where we mock-out most of the platform-specific code we're getting a null exception when new'ing up a ContentPage in `Page.On<Xamarin.Forms.PlatformConfiguration.iOS>()`

Changing this setter:
`<Setter Property="ios:Page.UseSafeArea" Value="true" />`
to this:
`<Setter Property="ios:Page.UseSafeArea">
  <OnPlatform x:TypeArguments="x:Boolean">
    <On Platform="iOS" Value="true" />
  </OnPlatform>
</Setter>`

works-around the issue, but I'm not sure if the final behavior will still be correct.

**Comment by SudeepSingh1502** [comment]
Try with xcode version 9.4

**Comment by pit-le-rouge** [comment]
1) now xcode 10 is installed.  And this problem don't come from xcode as it's working well with Foms 3.1
2) android version also crash.... so again, no link wioth xcode

**Comment by kensykora** [comment]
I'm seeing the same issue, and I tried @jalbertSyncroTech 's workaround, but it did not work for me.

Xamarin.Forms v3.2.0.839982
iOS 12.0
XCode 10.0

```
System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.PlatformConfiguration.iOSSpecific.Page.GetSafeAreaInsets (Xamarin.Forms.BindableObject element) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\PlatformConfiguration\iOSSpecific\Page.cs:134 
  at Xamarin.Forms.PlatformConfiguration.iOSSpecific.Page+<>c.<.cctor>b__27_0 (Xamarin.Forms.BindableObject bindable, System.Object oldValue, System.Object newValue) [0x00027] in D:\a\1\s\Xamarin.Forms.Core\PlatformConfiguration\iOSSpecific\Page.cs:70 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00120] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:625 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x0015b] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:417 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0003d] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:573 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:351 
  at Xamarin.Forms.Setter.Apply (Xamarin.Forms.BindableObject target, System.Boolean fromStyle) [0x000ba] in D:\a\1\s\Xamarin.Forms.Core\Setter.cs:65 
  at Xamarin.Forms.Style.ApplyCore (Xamarin.Forms.BindableObject bindable, Xamarin.Forms.Style basedOn) [0x0001e] in D:\a\1\s\Xamarin.Forms.Core\Style.cs:135 
  at Xamarin.Forms.Style.Xamarin.Forms.IStyle.Apply (Xamarin.Forms.BindableObject bindable) [0x0002b] in D:\a\1\s\Xamarin.Forms.Core\Style.cs:98 
  at Xamarin.Forms.MergedStyle.SetStyle (Xamarin.Forms.IStyle implicitStyle, System.Collections.Generic.IList`1[T] classStyles, Xamarin.Forms.IStyle style) [0x00156] in D:\a\1\s\Xamarin.Forms.Core\MergedStyle.cs:161 
  at Xamarin.Forms.MergedStyle.set_Style (Xamarin.Forms.IStyle value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\MergedStyle.cs:37 
  at Xamarin.Forms.VisualElement+<>c.<.cctor>b__249_6 (Xamarin.Forms.BindableObject bindable, System.Object oldvalue, System.Object newvalue) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:85 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00120] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:625 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x0015b] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:417 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0003d] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:573 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:99 
  at Xamarin.Forms.VisualElement.set_Style (Xamarin.Forms.Style value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:291 
  at MyApp.Views.GenericBackHeaderView.InitializeComponent () [0x00034] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/obj/Debug/netstandard2.0/Views/GenericBackHeaderView.xaml.g.cs:30 
  at MyApp.Views.GenericBackHeaderView..ctor () [0x00008] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/Views/GenericBackHeaderView.xaml.cs:17 
  at MyApp.Views.ContactUsView.InitializeComponent () [0x00056] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/obj/Debug/netstandard2.0/Views/ContactUsView.xaml.g.cs:38 
  at MyApp.Views.ContactUsView..ctor (MyApp.ViewModels.IViewModelBase contactViewModel) [0x00008] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/Views/ContactUsView.xaml.cs:16 
  at MyApp.Views.BossMasterDetailView.Navigate (MyApp.Enums.MenuType id, System.Boolean forceCreateNew) [0x000cd] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/Views/BossMasterDetailView.cs:55 
  at MyApp.Views.BossMasterView.MasterLayoutItemTappedHandler (MyApp.Enums.MenuType menuType) [0x00001] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/Views/BossMasterView.xaml.cs:80 
  at MyApp.Views.BossMasterLayoutNavigationItem.<.ctor>b__8_0 () [0x00008] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp/Views/BossMasterLayoutNavigationItem.cs:43 
  at Xamarin.Forms.Command+<>c__DisplayClass3_0.<.ctor>b__0 (System.Object o) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Command.cs:73 
  at Xamarin.Forms.Command.Execute (System.Object parameter) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Command.cs:107 
  at Xamarin.Forms.TapGestureRecognizer.SendTapped (Xamarin.Forms.View sender) [0x00018] in D:\a\1\s\Xamarin.Forms.Core\TapGestureRecognizer.cs:44 
  at Xamarin.Forms.Platform.iOS.EventTracker+<>c__DisplayClass13_0.<CreateRecognizerHandler>b__0 (UIKit.UITapGestureRecognizer sender) [0x0008f] in <66855f87e3b64ff38a3e1034f86d8129>:0 
  at UIKit.UITapGestureRecognizer+Callback.Activated (UIKit.UITapGestureRecognizer sender) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.0.0.15/src/Xamarin.iOS/UIKit/UIGestureRecognizer.cs:205 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.0.0.15/src/Xamarin.iOS/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.0.0.15/src/Xamarin.iOS/UIKit/UIApplication.cs:63 
  at MyApp.iOS.Application.Main (System.String[] args) [0x00000] in /Users/sykorkx/Workspace/BossDealer_XamarinMobile/src/MyApp.iOS/Main.cs:17
2018-09-26 16:14:56.087550-0500 MyApp.iOS[51247:7469879] Unhandled managed exception:
```

**Comment by pit-le-rouge** [comment]
I just tried to delete this line in my XAML:
ios:Page.UseSafeArea="true"

Once deleted, the page don't crash anymore.

SO, QUESTION: HOW TO MANAGE THE IOS SAFE AREA WITH XAMARIN.FORMS 3.2 ???

**Comment by PureWeen** [comment]
@pit-le-rouge  @kensykora @jalbertSyncroTech @wenyonk 

I haven't been able to recreate your issue. I've attached a repro that I'm using if you want to let me know how to see the crash

[SafeArea.zip](https://github.com/xamarin/Xamarin.Forms/files/2429932/SafeArea.zip)

That being said this PR
https://github.com/xamarin/Xamarin.Forms/pull/3811

Moves around the code that's throwing the NRE
https://github.com/xamarin/Xamarin.Forms/pull/3811/files#diff-d379831cb9daec7e5a1e569f9632d03eL70

And I feel like the order it will now execute with should resolve the issue you are seeing

**Comment by pit-le-rouge** [comment]
@PureWeen  
Simple to reproduce.
Take your example (safearea.zip) and add this sentence:
xmlns:ios="clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core"  
ios:Page.UseSafeArea="true">

in the APP.XAML (and not only in PAge1.xaml)

and then... BOUM

@samhouts 
Now, you have your example

**Comment by PureWeen** [comment]
@pit-le-rouge  if I'm understanding you correctly you're setting that value on the *<app* element?
That setting will only work on a Page and not on anything that isn't of type Page.  The PR that I've made will indeed "fix" the NRE but the result will just be that nothing happens

@jalbertSyncroTech I messed around with your scenario a little bit and there does look to be a bug where Global Styles aren't being applied for ContentPage types.

I was able to get it to work by placing this into my App.Xaml

```XAML
<Application xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="Triage.App"
             xmlns:ios="clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core"  
            >
	<Application.Resources>
         <ResourceDictionary>
            <Style TargetType="ContentPage" x:Key="pageStyles">
                <Setter Property="ios:Page.UseSafeArea" Value="true" />
            </Style>
        </ResourceDictionary>
	</Application.Resources>
</Application>
```

And then on the ContentPage applying the style

```XAML
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="Triage.Page1"
             Style="{StaticResource pageStyles}">
```

**Comment by jalbertSyncroTech** [comment]
@PureWeen I think this has always been the behavior by design: according to https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.style.targettype?view=xamarin-forms
"Developers should be aware that implicit styles are only applied to the specific type that is described by TargetType, and not to types that inherit from it."

**Comment by PureWeen** [comment]
@jalbertSyncroTech  I'm not quite following

```XAML
<Style TargetType="ContentPage>
                <Setter Property="ios:Page.UseSafeArea" Value="true" />
            </Style>
```

The TargetType I specified for the implicit style is ContentPage so it should just apply to a ContentPage without having to indicate a Style on the ContentPage

**Comment by jalbertSyncroTech** [comment]
@PureWeen The new Content Page template in Visual studio creates a new .xaml and .xaml.cs file pair that sub-class Content Page.  If you new up a ContentPage directly and attach it to the visual tree, which we do in some our our tests, then yes the implicit style should apply.  However, all of our production pages are actually sub-classes of ContentPage so the implicit style doesn't get applied to them.

**Comment by samhouts** [comment]
Should be closed by #3811. Thank you!

**Comment by lassana** [comment]
@samhouts  Any chances to get it back-ported to 3.2.0?
